kotlintest-assertions/src/main/kotlin/io/kotlintest/matchers/file/matchers.kt

```diff
@@ -233,3 +233,24 @@ fun containFileDeep(name: String): Matcher<Path> = object : Matcher<Path> {
       fileExists(value.toFile()), "File $name should exist in $value", "File $name should not exist in $value"
   )
 }
+
+fun Path.shouldContainFiles(vararg files: String) = this should containsAllFiles(files.asList())
+fun Path.shouldNotContainFiles(vararg files: String) = this shouldNot containsAllFiles(files.asList())
+fun containsAllFiles(files: List<String>) = object : Matcher<Path> {
+  override fun test(value: Path): Result {
+    val existingFiles = files.intersect(value.toFile().list().asIterable())
```

Comment:
I didn't check but can list() return null ?

Reply:
Yes, it can. Just saw that. I'll push a fix in a few minutes.